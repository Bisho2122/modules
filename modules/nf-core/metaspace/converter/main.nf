process METASPACE_CONVERTER {
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container 'docker.io/bwadie/metaspace_converter:latest'

    // container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
    //     'https://depot.galaxyproject.org/singularity/YOUR-TOOL-HERE':
    //     'biocontainers/YOUR-TOOL-HERE' }"

    input:
    val(ds_id)

    output:
    path("AnnData_${ds_id}.h5ad"), emit: AnnData_object
    path("SpatialData_${ds_id}.zarr"), emit: SpatialData_object
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    database_name = params.database_name
    database_version = params.database_version
    fdr = params.fdr
    use_tic = params.use_tic
    metadata_as_obs = params.metadata_as_obs

    template 'run_metaspace_converter.py'

    stub:
    """
    touch  AnnData_${ds_id}.h5ad
    touch  SpatialData_${ds_id}.zarr

    cat <<-END_VERSIONS > versions.yml
    ${task.process}:
        python: 3.11
        metaspace_converter: 1.1.1
    END_VERSIONS
    """

}
