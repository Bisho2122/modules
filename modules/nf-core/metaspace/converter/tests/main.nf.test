nextflow_process {

    name "Test Process METASPACE_CONVERTER"
    script "../main.nf"
    process "METASPACE_CONVERTER"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "metaspace"
    tag "metaspace/converter"

    test("metaspace_converter_test") {

        when {
            process {
                """
                input[0] = '2022-08-05_17h28m56s'
                """
            }
        }

        then {
            assertAll(
                { assert process.success }
            )
        }
    }
    test("metaspace_converter - stub") {

        options '-stub'
        when {
            process {
                """
                input[0] = '2022-08-05_17h28m56s'
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
