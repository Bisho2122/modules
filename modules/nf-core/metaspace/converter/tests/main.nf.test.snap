{
    "metaspace_converter - stub": {
        "content": [
            {
                "0": [
                    "AnnData_2022-08-05_17h28m56s.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "SpatialData_2022-08-05_17h28m56s.zarr:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,d5aa6645e814f32120fd2f6bf6b4414c"
                ],
                "AnnData_object": [
                    "AnnData_2022-08-05_17h28m56s.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "SpatialData_object": [
                    "SpatialData_2022-08-05_17h28m56s.zarr:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,d5aa6645e814f32120fd2f6bf6b4414c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-25T01:40:06.779340874"
    }
}